<a href="https://colab.research.google.com/github/MamontLara/Python_for_data_analyst_homeworks/blob/main/%22hw_6_2_%D0%A1%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D1%82%D0%B5%D1%81%D1%82%D0%B8%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашнє завдання 6: Статистичні тести

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [2]:
import numpy as np
import pandas as pd

In [3]:
df = pd.read_csv('drive/MyDrive/Dataset/data.csv')

In [4]:
data = np.array(df['data'])

In [5]:
type(data)

numpy.ndarray

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [6]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [7]:
from statsmodels.stats.weightstats import ztest

In [8]:
null_mean = 20
ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')

In [9]:
ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

Статистика для вибірки по z-критерію становить -0.995, а відповідне p-value дорівнює 0.840.

Оскільки p-value > 0.05, то ми не можемо відкинути нульову гіпотезу. Тобто немає статистично значущої різниці між середніми чеками за липень 2022 року та попередні місяці.

In [10]:
alpha = 0.05
if (p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [11]:
df_2 = pd.read_csv('drive/MyDrive/Dataset/data_2.csv')
data_2 = np.array(df_2['data'])

In [12]:
type(data_2)

numpy.ndarray

In [13]:
data_2.shape

(120,)

In [14]:
np.mean(data), np.mean(data_2)

(19.37755319670441, 19.37322276512519)

In [15]:
ztest(data, data_2, value=0)

(0.005246673456808504, 0.9958137794593356)

Р-значення є надзвичайно високим. Таке p-значення значно перевищує звичний рівень значущості (0.05). Це свідчить про те, що немає статистично значущих доказів для відхилення нульової гіпотези. Тобто, дані не вказують на істотну різницю між цими двома наборами.

Значення z-статистики близьке до нуля. Це означає, що середні значення двох наборів даних дуже схожі.

Отже немає істотної різниці між середніми значеннями цих наборів даних.